src/main/java/org/opensearch/jobscheduler/transport/JobParameterRequest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.jobscheduler.transport;
+
+import java.io.IOException;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.jobscheduler.spi.JobDocVersion;
+
+/**
+ * Request to extensions to parse a ScheduledJobParameter
+ */
+public class JobParameterRequest implements Writeable {
+
+    /**
+     * accessToken is the placeholder for the user Identity/access token to be used to perform validation prior to invoking the extension action
+     */
+    private final String accessToken;
+
+    /**
+     * jobSource is the index entry bytes reference from the registered job index
+     */
+    private final BytesReference jobSource;
+
+    /**
+     * id is the job Id
+     */
+    private final String id;
+
+    /**
+     * jobDocVersion is the metadata regarding this particular registered job
+     */
+    private final JobDocVersion jobDocVersion;
+
+    /**
+     * Instantiates a new Job Parameter Request
+     *
+     * @param accessToken the user identiy/access token that will be validated prior to triggering an extension action
+     * @param jobParser the parser obect to extract the jobSource {@link BytesReference} from
+     * @param id the job id
+     * @param jobDocVersion the job document version
+     * @throws IOException IOException when message de-serialization fails.
+     */
+    public JobParameterRequest(String accessToken, XContentParser jobParser, String id, JobDocVersion jobDocVersion) throws IOException {
+
+        // Extract jobSource bytesRef from xContentParser
+        XContentBuilder builder = XContentFactory.jsonBuilder();
+        builder.copyCurrentStructure(jobParser);
+        this.jobSource = BytesReference.bytes(builder);
+        this.accessToken = accessToken;
```

Comment:
I would leave the details of `accessToken` to @dbwiddis. 
My preference is to not care about security at this point and get the communication working and we can add security later. There might be lot more implications we have to consider like:
- How are we gating JS Rest APIs
- Who can modify a job parameter, job index (a.k.a any meta data) etc

Reply:
Agree the "details" will be implemented later but I think it's safe to put a placeholder String here for now as part of the "communication".  We're doing the same thing with the Rest Handler communication with a placeholder string and eventual authentication integration based on it.